pxp/transaction_status.md

```diff
@@ -32,32 +38,94 @@ entry must comply with the following JSON schema:
 | transaction_id | string | the id of the transaction the controller is interested in
 
 
-### Blocking Response
+### Response
 
-Transaction status responses must have the *results* entry complying with the
-following schema:
+The agent will respond with a [blocking responses][2] including a *stdout*
+entry in the *output* object that conforms to the following JSON schema:
 
 ```
 {
     "properties" : {
         "transaction_id" : { "type" : "string" },
-        "status" : { "enum" : [ "unknown", "success", "failure", "running" ] },
-        "stdout" : { "type" : "string" },
-        "stderr" : { "type" : "string" },
-        "exitcode" : { "type" : "integer" }
+        "status" : { "enum" : [ "unknown", "running", "success", "failure", "undetermined" ] },
+        "output" : {
+            "type" : "object",
+            "items" : {
+                "stdout" : { "description" : "should be a string, in case status is failure or undetermined, or the results in an action-specific format, otherwise" },
+                "stderr" : { "type" : "string" },
+                "exitcode" : { "type" : "number" },
+            },
+            "required" : [],
+            "additionalProperties" : false,
+        },
+        "metadata" : {
+            "type" : "object",
+            "items" : {
+                "execution_error" : { "type" : "string" },
+                "module" : { "type" : "string" },
+                "action" : { "type" : "string" },
+                "start" : { "type" : "string" },
+                "end" : { "type" : "string" },
+            },
+            "required" : ["module", "action", "start"],
+            "additionalProperties" : false
+        },
     },
     "required" : ["transaction_id", "status"],
-    "additionalProperties" : false
+    "additionalProperties" : false,
```

Comment:
But I see that we have similar occurrences all around the place. We should unify this in a separate PR.